pkg/kubelet/network/kubenet/kubenet_linux.go

```diff
@@ -281,6 +288,24 @@ func (plugin *kubenetNetworkPlugin) SetUpPod(namespace string, name string, id k
 		return err
 	}
 
+	// Put the container bridge into promiscuous mode to force it to accept hairpin packets.
+	// TODO: Remove this once the kernel bug (#20096) is fixed.
+	if plugin.wantPromisc && !plugin.havePromisc {
```

Comment:
@freehan do you mean just check every time?  I was attempting to minimize pod startup time somewhat, since this check is run every time any pod is started and would potentially shell out /sbin/ip each time which as a cost.  The downside of want/havePromisc is that if cbr0 is deleted underneath kubenet, promisc won't get reset.  But I'd rather wait until vishvanada/netlink gets promisc support to check every time, since a netlink call is a ton less expensive than a fork/exec.